lib/encode.js

```diff
@@ -96,6 +98,10 @@ export async function encode({
     // output uncompressed CBOR-LD
     suffix = cborg.encode(jsonldDocument);
   } else {
+    // lookup typeTable by id if needed
+    if(!isLegacy && !typeTable && typeTableLoader) {
+      typeTable = await typeTableLoader({registryEntryId});
+    }
```

Comment:
I think we should throw if both `typeTable` and `typeTableLoader` are set anywhere -- because that's most likely an error by the developer.